libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/model/Store.kt

```diff
@@ -53,26 +67,47 @@ internal open class Store {
 
     fun dispatch(action: Action) {
         if (isDispatching) {
-            logW(
-                "Cannot dispatch new actions during reducer processing. " +
-                    "Action dropped: ${action::class.java.simpleName}.",
-                "Store"
-            )
+            // Actions dispatched during reducer processing loop will be queued up
+            // and processed after current action processing finishes.
+            pendingActions.add(action)
```

Comment:
This depends on new events being dispatched right?

If, for some reason, there are no new actions. The last action will remain in the pending queue 🤔 . It won't happen often, but would be super confusing when it did.

Reply:
I've revered all changes to the Store in my latest update. This is no longer valid. Resolving this thread.